### ISSUE TITLE
Enable Infinispan async mode in external Infinispan

### ISSUE THREAD

**Issue by ahus1** [keycloak/keycloak-benchmark#524]
"async" as a new option in addition to "sync"

Run a first test setup - maybe in parallel?
* Share the .env file
* Sustained load test
* Failure test

**Comment by pruivo** [comment]
Variable `CROSS_DC_MODE` (default `SYNC`) controls the cross-site strategy. Change to `ASYNC` to enable asynchronous cross-site mode.

**Comment by ahus1** [comment]
@pruivo - as discussed today, please enable automatic state transfer one the secondary site becomes available again in our setup in our helm charts. Thanks!

**Comment by pruivo** [comment]
@ahus1 PR: https://github.com/keycloak/keycloak-benchmark/pull/529

**Comment by ahus1** [comment]
@pruivo - thank you, great stuff! PR is now merged.

**Comment by ahus1** [comment]
Conflict resolution to be configured for ASYNC (AFAIK only ASYNC has that feature):

**work cache, login failures, ...** 
* keep the entry (use newer entry?)

**sessions, single use, ...:** 
 * first approach: if there is a conflict, drop the entry
 * second approach: list activities that could lead to a conflict, and try to find better solutions

**Comment by ahus1** [comment]
Current default behavior in async: "sort by site name, first one wins the write"

**Comment by ahus1** [comment]
Async sustained load testing performed, `.env` and results need to be published here -> TODO for @mhajas 

Failure testing is still missing -> @kami619 / @martin-kanis

**Comment by mhajas** [comment]
### Sustained load results

TLDR: We were able to reproduce results for SYNC mode in ASYNC mode. 

### Details

`.env` file:
```
# GENERIC
TIMEOUT=1200 # 20 minutes in seconds - 300 may be too low if we need to wait for machines to be provisioned
ROSA_CLUSTER_NAME_1=mh-a
ROSA_CLUSTER_NAME_2=mh-b

# AURORA
AURORA_CLUSTER=mhajas
AURORA_REGION=eu-west-2

# ROUTE53
KC_CLIENT_URL=client.mh-a-mh-b-mtm4mtek.keycloak-benchmark.com
KC_HEALTH_URL_CLUSTER_1=primary.mh-a-mh-b-mtm4mtek.keycloak-benchmark.com
KC_HEALTH_URL_CLUSTER_2=backup.mh-a-mh-b-mtm4mtek.keycloak-benchmark.com

# INFINISPAN
KC_ISPN_NAMESPACE=mhajas-keycloak
CROSS_DC_MODE=ASYNC

# KEYCLOAK
KC_NAMESPACE_PREFIX='mhajas-'
KC_DATABASE=aurora-postgres
KC_DB_POOL_MIN_SIZE=15
KC_DB_POOL_MAX_SIZE=15
KC_DB_POOL_INITIAL_SIZE=15
KC_STORAGE=
KC_INSTANCES=6
KC_DISABLE_STICKY_SESSION='true'
KC_MEMORY_REQUESTS_MB=2048
KC_MEMORY_LIMITS_MB=3000
KC_HEAP_MAX_MB=2048
KC_HEAP_INIT_MB=1024
KC_CPU_REQUESTS=8
KC_CPU_LIMITS=8
KC_OTEL='true'
KC_CUSTOM_INFINISPAN_CONFIG_FILE=config/kcb-infinispan-cache-remote-store-config.xml
```

Command for running tests:
```
./benchmark.sh eu-west-2 \
  --scenario=keycloak.scenario.authentication.AuthorizationCode \
  --server-url=${KC_CLIENT_URL} \
  --users-per-sec=350 \
  --measurement=600 \
  --realm-name=realm-0 \
  --logout-percentage=100 \
  --users-per-realm=100000 \
  --ramp-up=100
```

Gatling result:
![image](https://github.com/keycloak/keycloak-benchmark/assets/13901995/a8130c51-60a6-489c-b363-ac30289109b5)

All results:
https://drive.google.com/file/d/10J9UbsZSnkC7npX5NESbBmhElOQQ6SjG/view?usp=sharing

**Comment by pruivo** [comment]
Gossip Router failover scenario here: https://drive.google.com/file/d/1ybobS3pdu5bdDwGd5yqlGRUiijaHMhlb/view?usp=drive_link

**Comment by ahus1** [comment]
Hi @pruivo - thank you for the test, these are great news. I'm closing this issue then.

I created one follow-up issue to review the existing conflict resolution strategies, see #551, as I assume there might be some low-hanging fruits to configure the existing behavior to cover some cases.

**Comment by pruivo** [comment]
Passive failing test also with expected results. With async, there is no impact on Keycloak requests. 
https://drive.google.com/file/d/1GMXE4QQfFU23QusfCG5JiOVJU1mXPc8q/view?usp=drive_link

**Comment by pruivo** [comment]
Continue with testing, it is missing the active failing.

**Comment by pruivo** [comment]
Keycloak route fail scenario: https://drive.google.com/file/d/1m4s6zLBWG0KVmIhfgCX1FcmbdWp_e3WX/view?usp=drive_link

![keycloak-route-fail-scenario](https://github.com/keycloak/keycloak-benchmark/assets/1492066/f8d4c8c5-2700-47a2-8d90-4a1bbe1b4f92)

**Comment by pruivo** [comment]
Let me know if I miss some scenario.

**Comment by ahus1** [comment]
@pruivo - thank you for testing these scenarios, this looks good to me.
